apps/arena/lib/arena/game_updater.ex

```diff
@@ -314,20 +314,24 @@ defmodule Arena.GameUpdater do
   # End Zone Callbacks
   ##########################
 
-  def handle_info(
-        {:to_killfeed, killer_id, victim_id},
-        %{game_state: game_state, game_config: game_config} = state
-      ) do
+  def handle_info({:to_killfeed, killer_id, victim_id}, %{game_state: game_state, game_config: game_config} = state) do
+    last_id = game_state.last_id + 1
     entry = %{killer_id: killer_id, victim_id: victim_id}
     victim = Map.get(game_state.players, victim_id)
 
+    on_death_effect =
+      Enum.find(game_config.effects, fn effect -> effect.name == victim.aditional_info.on_death_effect end)
+
+    killer =
+      Map.get(game_state.players, killer_id)
+      |> Player.apply_on_death_effect(on_death_effect, last_id)
+      |> update_in([:aditional_info, :kill_count], fn count -> count + 1 end)
+
     amount_of_power_ups = get_amount_of_power_ups(victim, game_config.power_ups.power_ups_per_kill)
 
     state =
-      update_in(state, [:game_state, :killfeed], fn killfeed -> [entry | killfeed] end)
-      |> update_in([:game_state, :players, killer_id, :aditional_info, :kill_count], fn count ->
-        count + 1
-      end)
+      put_in(state, [:game_state, :players, killer.id], killer)
```

Comment:
Maybe we could do something here like the `:update_game` callback?
Something like:
```elixir
state =
  state
  |> update_killer_player()
  |> update_killfeed()
``` 
that way we could isolate those behaviors in separate functions and is easier to code/follow.

Reply:
Made a refactor for this https://github.com/lambdaclass/mirra_backend/pull/389/commits/84b198e65cd780f63953eafb1a16ce56bd544e66